lib/age.c

```diff
@@ -162,7 +162,11 @@ void agecheck (/*@null@*/const struct spwd *sp)
 		return;
 	}
 
-	remain = sp->sp_lstchg + sp->sp_max - now;
+	remain = sp->sp_lstchg - now;
```

Comment:
Is there any way that the - now can lead to overflow?  Would probably require time(2) to return -1, and sp->sp_lstchg to already be max?

Reply:
Yes, I assume that time is positive, i.e. the system time is properly set. We can protect against it though.

Reply:
Updated the PR. This should do the trick. In this scenario, the difference is larger than LONG_MAX, i.e. the warn period cannot be reached. So we can directly call return.

Reply:
Thanks, yes, I think that's good.